src/MapWindow/GlueMapWindowItems.cpp

```diff
@@ -49,7 +49,16 @@ GlueMapWindow::ShowMapItems(const GeoPoint &location,
   const MoreData &basic = CommonInterface::Basic();
   const DerivedInfo &calculated = CommonInterface::Calculated();
 
-  auto range = visible_projection.DistancePixelsToMeters(Layout::GetHitRadius());
+  bool only_landable = false;
+  int rangedisplaypercent = settings.item_list.range_of_nearest_map_items_in_percent_of_displaysize;
+  auto range = 0.0;
+  if(rangedisplaypercent == 0) range = visible_projection.DistancePixelsToMeters(Layout::GetHitRadius());
+  else {
+    range = visible_projection.DistancePixelsToMeters(Layout::min_screen_pixels/2*rangedisplaypercent/100);
```

Comment:
Should use SI units. The `range` variable appears to be in kilometres, and that is just asking for bugs.